conans/test/integration/toolchains/apple/test_xcodedeps.py

```diff
@@ -196,3 +196,85 @@ def package_info(self):
     lib_a_xcconfig = client.load("conan_lib_a_lib_a_release_x86_64.xcconfig")
 
     assert "lib_a_frameworkdir" in lib_a_xcconfig
+
+
+@pytest.mark.skipif(platform.system() != "Darwin", reason="Only for MacOS")
+def test_xcodedeps_cppinfo_requires():
+
+    """
+    lib_a: has four components cmp1, cmp2, cmp3, cmp4
+    lib_b --> uses libA cmp1 so cpp_info.requires = ["lib_a::cmp1"]
+    lib_c --> uses libA cmp2 so cpp_info.requires = ["lib_a::cmp2"]
+    consumer --> libB, libC
+    """
+    client = TestClient()
+    lib_a = textwrap.dedent("""
+        from conan import ConanFile
+        class lib_aConan(ConanFile):
+            name = "lib_a"
+            version = "1.0"
+            settings = "os", "compiler", "build_type", "arch"
+            def package_info(self):
+                self.cpp_info.components["cmp1"].includedirs = ["include"]
```

Comment:
That is just used to create the components, the test does not test that include directory exists, because that is added in the own component file: `conan_lib_a_cmp1_release_x86_64.xcconfig` it tests that `conan_lib_b_lib_b.xcconfig` does not include the xcconfig of the component, I'll add some comments to clarify the test

Reply:
Yes, it appears in the file from the four components:

- `conan_lib_a_cmp1_release_x86_64.xcconfig`
- `conan_lib_a_cmp2_release_x86_64.xcconfig`
- ...

like:

```
HEADER_SEARCH_PATHS_lib_a_cmp1[config=Release][arch=x86_64][sdk=*] = ".../package/4b92821f1af51ada4198018b78c8b197b32c6c18/include"
```
Then, that variable is inherited in the `conan_lib_a_cmp1.xcconfig` like:

```
#include "conan_lib_a_cmp1_release_x86_64.xcconfig"

HEADER_SEARCH_PATHS = $(inherited) $(HEADER_SEARCH_PATHS_lib_a_cmp1)
```

and then, the lib_b xcconfig includes `#include "conan_lib_a_cmp1.xcconfig"`

```
// Conan XcodeDeps generated file for lib_b::lib_b
// Includes all configurations for each dependency

// Includes for lib_a::cmp1 dependency
#include "conan_lib_a_cmp1.xcconfig"

#include "conan_lib_b_lib_b_release_x86_64.xcconfig"
```

But in this file we have no trace of include directories, just variables that inherit from that value in `conan_lib_a_cmp1_release_x86_64.xcconfig` so I can't assert that value

Reply:
ok. nevermind.